pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -105,3 +114,30 @@ func getOAuthTokenCacheTTL(t time.Time) time.Duration {
 
 	return ttl
 }
+
+// hasIDTokenExpired checks if the exp field in the ID token has expired. It does not validate the ID token.
+func hasIDTokenExpired(token *login.UserAuth) (bool, error) {
```

Comment:
I would suggest to change this function to something like `func getIDTokenExpiry(token *login.UserAuth) (time.Time, error)`, because this function could be reused to calculate the cache TTL properly.

Explanation: The tokens are cached to improve performance and currently the access token's expiry is set as a TTL, considering that the ID token's expiry could be earlier than the AT expiry I think the cache TTL should be calculated with the following formula: `min(access_token expiry, id_token expiry)`.
https://github.com/grafana/grafana/pull/69829/files#diff-ecd19dccd5583801dfd8752e347c08427fa81deba3285c6e7e17ccba016cd382R80